lib/plugins/aws/package/compile/events/apiGateway/lib/requestValidator.test.js

```diff
@@ -0,0 +1,404 @@
+'use strict';
+
+const chai = require('chai');
+const runServerless = require('../../../../../../../../tests/utils/run-serverless');
+const fixtures = require('../../../../../../../../tests/fixtures');
+
+const expect = chai.expect;
+chai.use(require('chai-as-promised'));
+
+describe('#compileRequestValidators()', () => {
+  let globalServerless;
+
+  before(() => {
+    globalServerless = runServerless({
+      cwd: fixtures.map.requestSchema,
+      cliArgs: ['package'],
+    });
+  });
+
+  after(() => {
+    fixtures.cleanup();
+  });
+
+  describe(' reusable schemas ', () => {
+    it('Should process schema from apiGateway provider, full config', () =>
+      globalServerless.then(serverless => {
+        const cfResources = serverless.service.provider.compiledCloudFormationTemplate.Resources;
+        const naming = serverless.getProvider('aws').naming;
+        const modelLogicalId = naming.getModelLogicalId('TestModel');
+        const modelResource = cfResources[modelLogicalId];
+        expect(modelResource).to.deep.equal({
+          Type: 'AWS::ApiGateway::Model',
+          Properties: {
+            ContentType: 'application/json',
+            Description: 'Test Description',
+            Name: 'TestModel',
+            RestApiId: {
+              Ref: 'ApiGatewayRestApi',
+            },
+            Schema: {
+              $schema: 'http://json-schema.org/draft-04/schema#',
+              definitions: {},
+              properties: {
+                id: {
+                  pattern: '[0-9]+',
+                  title: 'ID for object',
+                  type: 'number',
+                },
+              },
+              required: ['id'],
+              title: 'Test Validation Schema',
+              type: 'object',
+            },
+          },
+        });
+      }));
```

Comment:
Really great tests! Thank you!

I'm always trying to minimize dependency on implementation details, so usually I do not confirm on every single property but on one or few crucial. So e.g. some cleanup doesn't necessarily break the tests.

(just a note for future, I do not expect you to update above)